src/components/big-interactive-pages/editor.tsx

```diff
@@ -145,6 +163,38 @@ export default function Editor({ persistenceState, cookies }: EditorProps) {
 	const screenContainer = useRef<HTMLDivElement>(null);
 	const screenControls = useRef<HTMLDivElement>(null);
 
+	const [sessionId] = useState(nanoid());
+
+	useEffect(() => {
+		console.log('New session started with ID:', sessionId);
+		const channel = new BroadcastChannel('session_channel');
+		channel.onmessage = (event) => {
+			if (event.data.type === 'SESSION_UPDATE' && event.data.sessionId !== sessionId) {
+				console.log('New session detected:', event.data.sessionId);
+			}
+		};
+		channel.postMessage({ type: 'SESSION_UPDATE', sessionId });
+		return () => channel.close();
+	}, [sessionId]);
+
+	useEffect(() => {
+
+		const handleUnload = () => {
+			localStorage.removeItem('lastSavedSessionId');
```

Comment:
why is it important to delete this local session id on unload?  why is this necessary at all?

Reply:
The **lastSavedSessionId** is removed on page unload to ensure that each session starts fresh, preventing the website from mistakenly displaying an overwrite warning when users return to the editor.

Reply:
it's replicate how SessionStorage works, it destory on session / tab end, problem with SessionStorage on our case is that it doesn't communicate between tabs, so we need used LocalStorage with this manual remove on unload.